fix: segfault when rank is too big

AndreCostaaa opened the pull request:
Fixes #1  by checking the percentil rank and ~adapting it~ throwing an error if it's out of bounds.
~Additionally log a message so the user is aware of the problem~

~Also added a test to check that the rank value imported is correctly adapted if it's out of bounds~

```bash
$ tail imatinib.tqf
                  <rank>25</rank>
                  <rank>50</rank>
                  <rank>75</rank>
                  <rank>90</rank>
                  <rank>5000</rank>
              </ranks>
            </percentilesTraits>
        </request>
    </requests>
</query>

$ ./tucucli -d drugs -i imatinib.tqf -o output.xml
[24-12-16 18:13:35.903] 109836 info: ********************************************************
[24-12-16 18:13:35.903] 109836 info: Tucuxi console application is starting up...
[24-12-16 18:13:35.903] 109836 info: Drugs directory : drugs
[24-12-16 18:13:35.904] 109836 info: Input file : imatinib.tqf
[24-12-16 18:13:35.904] 109836 info: Output file name : output.xml
[24-12-16 18:13:35.904] 109836 info: QueryLogs directory : 
[24-12-16 18:13:35.904] 109836 info: Data file directory : 
[24-12-16 18:13:35.904] 109836 info: Tqf copy output file : 
[24-12-16 18:13:35.907] 109836 error: Error, see details : 
<query><requests><request><percentilesTraits><ranks><rank> contains an invalid value : 5000

[24-12-16 18:13:35.908] 109836 info: The response XML file was successfully generated
[24-12-16 18:13:35.908] 109836 info: Tucuxi console application is exiting...
``` 

@ythoma please let me know if this is enough or if you think we should double check the values where they're actually used

AndreCostaaa committed:
test: add test for query import with invalid rank

AndreCostaaa committed:
fix(queryimport): check percentil rank range on import

ythoma commented:
Thanks for that. Actually the issue can also arise if tucuxi-core is used directly, without XML import. Therefore we would also need to handle that case elsewhere. As the entry point for all computations is ComputingComponent, I think we could check the percentiles ranks in the corresponding percentiles function. We could add a ComputingStatus enum value like OutOfBoundsPercentileRank that could be returned for such error in the input.

AndreCostaaa committed:
feat: check percentile ranks before starting computation

AndreCostaaa committed:
test: add computing tests with invalid ranks

AndreCostaaa committed:
fix: throw error on query import if rank is out of bounds

AndreCostaaa commented:
> Thanks for that. Actually the issue can also arise if tucuxi-core is used directly, without XML import. Therefore we would also need to handle that case elsewhere. As the entry point for all computations is ComputingComponent, I think we could check the percentiles ranks in the corresponding percentiles function. We could add a ComputingStatus enum value like OutOfBoundsPercentileRank that could be returned for such error in the input.

Sounds good to me. Implemented on a86ea37

Also added the corresponding test

AndreCostaaa committed:
fix: remove unused variable

AndreCostaaa commented:
Just updated the PR so it documents correctly the work done here
